osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -309,6 +308,8 @@ private void load(AudioManager audio, DialogOverlay dialog, OsuColour colours, M
                     }
                 });
             }
+            Carousel.FilterControl = FilterControl;
```

Comment:
Giving components references to other components which reside in an entirely different branch of the draw hierarchy is generally not how things should be done. If anything the song select screen should be coordinating the count hand-off rather than this.